app/services/customers/create_service.rb

```diff
@@ -78,47 +79,56 @@ def create(**args)
 
-      create_result = PaymentProviderCustomers::CreateService.new(customer).create_or_update(
-        customer_class: PaymentProviderCustomers::StripeCustomer,
-        payment_provider_id: customer.organization.stripe_payment_provider&.id,
-        params: billing_configuration,
-      )
-      create_result.throw_error unless create_result.success?
+      customer.update!(payment_provider: billing_configuration[:payment_provider]) if is_api
+
+      create_or_update_provider_customer(customer, billing_configuration)
     end
 
     def handle_api_billing_configuration(customer, params, new_customer)
-      unless params.key?(:billing_configuration)
-        create_billing_configuration(customer) if new_customer
-        return
-      end
+      return unless params.key?(:billing_configuration)
 
       billing_configuration = params[:billing_configuration]
 
-      unless billing_configuration[:payment_provider] == 'stripe'
+      if new_customer
+        create_billing_configuration(customer, billing_configuration, is_api: true)
+        return
+      end
+
+      unless billing_configuration[:payment_provider] == 'stripe' ||
+             billing_configuration[:payment_provider] == 'gocardless'
         customer.update!(payment_provider: nil)
         return
       end
 
-      customer.update!(payment_provider: 'stripe')
+      customer.update!(payment_provider: billing_configuration[:payment_provider])
       create_or_update_provider_customer(customer, billing_configuration)
     end
 
     def create_or_update_provider_customer(customer, billing_configuration = {})
-      create_result = PaymentProviderCustomers::CreateService.new(customer).create_or_update(
-        customer_class: PaymentProviderCustomers::StripeCustomer,
-        payment_provider_id: customer.organization.stripe_payment_provider&.id,
-        params: billing_configuration,
-        async: !(billing_configuration || {})[:sync],
-      )
+      create_result = case billing_configuration[:payment_provider]
+                      when 'stripe'
+                        PaymentProviderCustomers::CreateService.new(customer).create_or_update(
+                          customer_class: PaymentProviderCustomers::StripeCustomer,
+                          payment_provider_id: customer.organization.stripe_payment_provider&.id,
+                          params: billing_configuration,
+                          async: !(billing_configuration || {})[:sync],
+                        )
+                      when 'gocardless'
+                        PaymentProviderCustomers::CreateService.new(customer).create_or_update(
+                          customer_class: PaymentProviderCustomers::GocardlessCustomer,
+                          payment_provider_id: customer.organization.gocardless_payment_provider&.id,
+                          params: billing_configuration,
+                          async: !(billing_configuration || {})[:sync],
+                        )
+      end
```

Comment:
I'm proposing something like this to make it a bit shorter:
```suggestion
      provider_class = case billing_configuration[:payment_provider]
                                 when 'stripe'
                                   PaymentProviderCustomers::StripeCustomer
                                 when 'gocardless'
                                   PaymentProviderCustomers::GocardlessCustomer
                                 end
                                 
      create_result = PaymentProviderCustomers::CreateService.new(customer).create_or_update(
        customer_class: provider_class,
        payment_provider_id: customer.organization.payment_provider(billing_configuration[:payment_provider])&.id,
        params: billing_configuration,
        async: !(billing_configuration || {})[:sync],
      )
```

The `provider_class` initialization could even be performed in a different method